duckduckgo/macos-browser
# Set main frame URL on link protection for redirect detection

Task/Issue URL: https://app.asana.com/0/1163321984198618/1202227967754334/f
Tech Design URL:
CC: @bwaresiak 

**Description**:
This PR integrates BSK changes to detect redirects in the link protection module and skips rewriting if a redirect is detected.



**Steps to test this PR**:
1. Visit bing.com and search "t-shirts bombas"
2. Click the first ad link. You should be taken to the destination website
1. Visit https://privacy-test-pages.glitch.me/privacy-protections/query-parameters/
3. Ensure links display correct "expected" value

**Testing checklist**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
